hg-fast-export.sh

```diff
@@ -3,7 +3,7 @@
 # Copyright (c) 2007, 2008 Rocco Rutte <pdmef@gmx.net> and others.
 # License: MIT <http://www.opensource.org/licenses/mit-license.php>
 
-ROOT="$(dirname "$(which "$0")")"
+ROOT="$(dirname "$(greadlink -f "$(which "$0")" 2> /dev/null || readlink -f "$(which "$0")")")"
```

Comment:
You could do this in two stages to make it a bit simpler to understand what's happening and why:

```shell
READLINK="readlink"
if command -v greadlink > /dev/null; then
  READLINK="greadlink" # Prefer greadlink over readlink
fi
ROOT="$(dirname "$($READLINK -f "$(which "$0")")")"
```